hpx/runtime/actions/basic_action.hpp

```diff
@@ -271,11 +271,10 @@ namespace hpx { namespace actions
         ///////////////////////////////////////////////////////////////////////
         typedef typename traits::is_future<local_result_type>::type is_future_pred;
 
-        template <typename LocalResult>
         struct sync_invoke
         {
             template <typename IdOrPolicy, typename ...Ts>
-            HPX_FORCEINLINE static LocalResult call(
+            HPX_FORCEINLINE static local_result_type call(
                 boost::mpl::false_, launch policy,
```

Comment:
Might have been a good idea to replace `boost::mpl::true_/false_` here as well - all the rest of the file has been switched to `<type_traits>`.